### DISCUSSION TITLE
Batch Script

### DISCUSSION THREAD

**Discussion by starbez** [smacke/subsync#158] [Q&A] [answered]
Can someone help me make a batch script?
what I want is a folder inside which are three folders one is "Reference" mkv and the other "Subtitles" and last "Output".
Reference folder has mkv like 1.mkv 2.mkv 3.mkv..... and Subtitles folder has 1.srt 2.srt 3.srt, each file should correspond with the file of same name 1.mkv - 1.srt, 2.mkv - 2.srt, 3.mkv - 3.srt........ and lastly save in "output" folder as 1.srt 2.srt 3.srt. I run batch script and it will synced all accordingly.
Please.

**Comment by tonyhsie** [comment, marked as answer]
I've written a tool for do the same job. It uses both ffsubsync & alass ( https://github.com/kaegi/alass ) to sync the same reference files and then generates double output files by ffsubsync & alass separately.

I don't use batch file since there are too many limits in batch file.

For using my tool, you don't have to rename the resource files or subtiltes files first.
Just drag-n-drop two files (one reference file & one subtitle file), or one directory (reference & subtitle files), or two directories (one reference directory & one subtitle directory) on the tool, the synchronization will auto start.

You don't have ffsubsync & alass both installed, the tool still works without one of them.

You may give it a try.
[Alass FFsubsync Helper.zip](https://github.com/smacke/ffsubsync/files/8494385/Alass.FFsubsync.Helper.zip)

**Comment by starbez** [comment]
At first I was a little worried after seeing .exe 😅 but thank god i ran it. it saved me hours of work. 😄
Thanks.

**Comment by tehniemer** [comment]
@tonyhsie Thanks for sharing this tool, it seems like it'll really streamline synchronization. I've used it by dragging directories with multiple reference and subtitle files onto the tool and it seems to synchronize some of the files but not all. If I drag the individual reference and subtitle files onto the tool it seems to work fine.

I took a look at your alass repo and didn't see Alass FFsubsync Helper on there so I'm not sure where to open an issue.

**Reply by tonyhsie** [reply]
[Alass FFsubsync Helper.v220611.zip](https://github.com/smacke/ffsubsync/files/9630831/Alass.FFsubsync.Helper.v220611.zip)
I've uploaded a new version of "Alass FFsubsync Helper", you may try if your problem is solved or not.

**Comment by smacke** [comment]
Folks, appreciate the discussion on various extensions for batch mode here. Thanks @tonyhsie for working on this helper. Just a reminder to everyone who sees this to please exercise caution when running downloaded code (ffsubsync isn't any exception of course).

Just a heads up that you can also script this with bash as follows (using 10 movies as an example):

```bash
for i in {1..10}; do
    ffs reference/$i.mkv -i in/$i.srt -o out/$i-synced.srt
done
```

Please also see #144 for an example of using ffsubsync to sync multiple subtitle files against the same reference on both Windows and *nix CLI flavors.

**Reply by tonyhsie** [reply]
I can't agree with you more about the exercise caution.

I just want to simplify the routine work of "renaming files" or "setting arguments", so I wrote the helper several months ago.

Thanks @smacke  for the creation of ffsubsync, it really helps a lot!

**Reply by GreenPhenz** [reply]
Is there a way to get your helper to work on mac?

**Reply by tonyhsie** [reply]
Sorry, there is no mac version since i'm not familiar with mac.

**Reply by GreenPhenz** [reply]
@smacke Could you explain what the reference/, in/, and out/ is supposed to be replaced with? Also is the output file already supposed to be created or can the program automatically make one (and if so how can I direct which folder for it to go in?)?

**Reply by GreenPhenz** [reply]
> Sorry, there is no mac version since i'm not familiar with mac.


@tonyhsie wow that was a really quick reply on a year old thread and rip.

**Reply by speelgoedauto2** [reply]
Im in the same boat guys.. i really dont know how to sync batches like tv shows, can someone help a bit?

**Reply by DJHanceNL** [reply]
for %i in (*.mp4) do ffs %~ni.mp4 -i %~ni.srt -o %~ni_sync.srt


in terminal

or change mp4 to .mkv.

now you only got to remove the bad srt and replace with the _sync ones

**Comment by starbez** [comment]
Ah, how naive I was.
